### ISSUE TITLE
Transclusion of a heading within an org file won't work for ID-type links 

### ISSUE THREAD

**Issue by alexiabw** [nobiot/org-transclusion#237]
For any org file containing an ID, the content of headings referenced by file type links can be transcluded, but this doesn't happen for ID-type links.

(Following the link itself works, so the link isn't broken.)

---

`[[file:somenode.org::*Some heading][Some heading]]` 

Is transcluded with no problems

`[[id:4a3fc3b3-caef-47d7-8e2d-ee0d246b63de::*Some heading][Some heading]]`

Causes the error 
> No content found with "id:4a3f...*Some heading". Check the link at point...

I am using Doom (on Linux), in case it could be a related issue.

**Comment by nobiot** [comment]
Can you try ID without the “::*Sime heading” part, please? 

[[id:4a3fc3b3-caef-47d7-8e2d-ee0d246b63de][Some heading]]

**Comment by nobiot** [comment]
@axbweiss 

This is a [custom feature ](https://github.com/doomemacs/doomemacs/blob/master/modules/lang/org/config.el#L624-L625)that Doom Emacs has added, and has yet to be incorporated in the upstream Org. 

I am not able to test this with Doom-specific syntax because I am not a Doom user. But perhaps you can add a customizing by using the same technique discussed in #160. I am not against adding it to the code base if this can be done cleanly. Will you be willing to do try?

**Comment by axbweiss** [comment]
Thank you @akashpal-21! I can confirm that the function is added to the `org-transclusion-add-functions` variable and that the original ID function was removed from it. The custom function works for ID-type links without any headings referenced, i.e. `[[id:4a3fc3b3-caef-47d7-8e2d-ee0d246b63de][File]]`. However, when attempting `[[id:4a3fc3b3-caef-47d7-8e2d-ee0d246b63de::*Some heading][Some heading]]` there is a new error: "No match for fuzzy expression:*Some"

I'll look into this these days; it's probably Doom-related, since it works on your end.

**Comment by nobiot** [comment]
@axbweiss @akashpal-21 

An alternative is to use `org-roam-db-extra-links-exclude-keys`. You can add your own custom property, eg "ROAM_EXCLUDE" to this list. You then add "ROAM_EXCLUDE" to the node with a value "t" (or whatever). 

This is probably easier given the error you are experiencing, @axbweiss .

It works on my end.

![image](https://github.com/nobiot/org-transclusion/assets/12507865/a2e4b64f-6279-4948-8d06-486c3f62d62a)

![image](https://github.com/nobiot/org-transclusion/assets/12507865/6360c6f7-9ccf-40ce-9cbb-eba0fa9ae089)

**Comment by axbweiss** [comment]
> Can you confirm that you are using the latest version of the function -- I fixed the problem when the translation failed if it had a space in it -- i did a silly mistake when parsing the link earlier.

@akashpal-21 Yes, I was using that function and it actually does work!! Thank you so much for your help! I think it'd be a great extension for Doom users, or anyone who has search syntax set up for ID links.

#52 was the actual issue; while testing the function, I'd had the source for the transclusion open in the same buffer as well. It still slips my mind to close source buffers before transcluding.

> An alternative is to use `org-roam-db-extra-links-exclude-keys`.

Thank you @nobiot! I'll be sure to remember that functionality for the future.

**Comment by axbweiss** [comment]
I've also added your workaround from #52 to my config, nobiot:

`(advice-remove 'org-link-search '+org--recenter-after-follow-link-a)`

It works. Thank you for this as well!

**Comment by nobiot** [comment]
@axbweiss Will you close the issue, or do you prefer to keep it open?

**Comment by axbweiss** [comment]
I opted to keep it open in case @akashpal-21 would like to make a merge request to add their function as an extension. If they'd rather not, I'll close the issue but might still make a request myself (crediting them, of course)

**Comment by jarofromel** [comment]
This is also org-mode feature not  doom emacs only  (at least from org-mode 9.7 version). I came to this issue because I also tried the tranclusion with the link `[[id:<org-mode ID>::*<org-mode heading>]]` without success.

For reference [org-mode 9.7](https://orgmode.org/Changes.html):

   > New option `org-id-link-consider-parent-id` to allow `id: links` to parent headlines

   > Storing a link with point at "Child 1" will produce a link `<id:abc::*Child 1>`, which precisely links to the "Child 1" headline even though it does not have its own ID.

Also the terse description in the [manual](https://orgmode.org/manual/External-Links.html):

   > id
   > `id:B7423F4D-2E8A-471B-8810-C40F074717E9`
   > `id:B7423F4D-2E8A-471B-8810-C40F074717E9::*task` (headline search)

**Comment by nobiot** [comment]
> This is also org-mode feature not doom emacs only (at least from org-mode 9.7 version). 

I realize Org 9.7 has added this new feature at some point. 9.7 was released in June 2024; the original conversation happened 1-2 months prior to it in April 2024. 

I am not certain when I can work to adjust `org-transclusion`. I'd love support here.

**Comment by nobiot** [comment]
@akashpal-21 , I am considering a more generic solution: 823982357fb8d33e4b3fcac5ba4234624204a3cd. More later :)

**Comment by nobiot** [comment]
For got to mention pr: #267
